toolchain/check/import_ref.cpp

```diff
@@ -2443,6 +2514,67 @@ static auto TryResolveTypedInst(ImportRefResolver& resolver,
   return ResolveResult::Done(impl_const_id, new_impl.first_decl_id());
 }
 
+static auto TryResolveTypedInst(ImportRefResolver& resolver,
+                                SemIR::RequireImplsDecl inst) -> ResolveResult {
+  const auto& import_require =
+      resolver.import_require_impls().Get(inst.require_impls_id);
+
+  // Load dependent constants.
+  auto parent_scope_id =
+      GetLocalNameScopeId(resolver, import_require.parent_scope_id);
+  auto generic_data = GetLocalGenericData(resolver, import_require.generic_id);
+  auto self_const_id = GetLocalConstantId(resolver, import_require.self_id);
+  auto facet_type_const_id =
+      GetLocalConstantId(resolver, import_require.facet_type_inst_id);
+
+  if (resolver.HasNewWork()) {
+    return ResolveResult::Retry();
+  }
+
+  // Make the decl and structure with placeholder values to be filled in.
+  SemIR::RequireImplsDecl require_decl = {
+      .require_impls_id = SemIR::RequireImplsId::None,
+      .decl_block_id = SemIR::InstBlockId::Empty};
+  auto require_decl_id = AddPlaceholderImportedInst(
+      resolver, import_require.decl_id, require_decl);
+  auto require_impls_id = resolver.local_require_impls().Add(
+      {.self_id = SemIR::TypeInstId::None,
+       .facet_type_inst_id = SemIR::TypeInstId::None,
+       .facet_type_id = SemIR::FacetTypeId::None,
+       .decl_id = require_decl_id,
+       .parent_scope_id = SemIR::NameScopeId::None,
+       .generic_id = MakeIncompleteGeneric(resolver, require_decl_id,
+                                           import_require.generic_id)});
+
+  // Write the RequireImplsId into the RequireImplsDecl.
+  require_decl.require_impls_id = require_impls_id;
+  auto require_decl_const_id =
+      ReplacePlaceholderImportedInst(resolver, require_decl_id, require_decl);
+
+  // Fill in the RequireImpls structure.
+  auto& new_require = resolver.local_require_impls().Get(require_impls_id);
+  new_require.self_id = resolver.local_types().GetAsTypeInstId(
+      AddLoadedImportRef(resolver, SemIR::TypeType::TypeId,
+                         import_require.self_id, self_const_id));
+  new_require.facet_type_inst_id =
+      resolver.local_types().GetAsTypeInstId(AddLoadedImportRef(
+          resolver, SemIR::TypeType::TypeId, import_require.facet_type_inst_id,
+          facet_type_const_id));
```

Comment:
```suggestion
  new_require.self_id = AddLoadedImportRefForType(resolver,
                         import_require.self_id, self_const_id);
  new_require.facet_type_inst_id =
      AddLoadedImportRefForType(
          resolver, import_require.facet_type_inst_id,
          facet_type_const_id);
```

Per comment about #6352 conflict. Sorry, I had been assuming either you would notice or I'd clean up later -- hadn't been expecting you'd change the other places doing this.